src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -1558,7 +1558,8 @@ std::string ClusterTaskManager::GetBestSchedulableNode(const internal::Work &wor
                                                        bool *is_infeasible) {
   // If the local node is available, we should directly return it instead of
   // going through the full hybrid policy since we don't want spillback.
-  if (work.grant_or_reject && !force_spillback && IsLocallySchedulable(work.task)) {
+  if ((work.grant_or_reject || work.is_selected_based_on_locality) && !force_spillback &&
```

Comment:
Would it make sense to unify grant_or_reject and is_selected_based_on_locality?

Reply:
They are the same if the node is available. But if the node is not available, their behavior is different, one is spill, the other is reject. I think it might be clearer to separate them.